lib/matplotlib/path.py

```diff
@@ -1042,3 +1203,248 @@ def get_path_collection_extents(
+        fits into the extents when stroked with a width of *markeredgewidth*.
+    vinfo : `.VertexInfo`
+        Information about the corner or cap at one vertex.
+    markeredgewidth : `float`
+        The width of the stroke being drawn.
+    joinstyle : {'miter', 'bevel', 'round'}
+        How line corners are to be drawn.
+    capstyle : {'butt', 'round', 'projecting'}
+        How line ends are to be drawn.
+
+    Notes
+    -----
+    Implemented by wrapping `._stroke_x_overflow`. This function checks which
+    direction the corner (or cap) is pointing, then for each side of *extents*
+    that might need padding, it rotates the corner to point in the positive x
+    direction and calls `._stroke_x_overflow` to get the padding.
+    """
+    xmin = 0
+    ymin = 1
+    xmax = 2
+    ymax = 3
+    # now for each direction (up/down/left/right), convert the absolute
+    # incidence angle into the incidence angle relative to that respective side
+    # of the bbox, and see if the stroked vertex expands the extents...
+    x, y = vinfo.apex
+    if np.cos(vinfo.incidence_angle) > 0:
+        incidence_angle = vinfo.incidence_angle + np.pi/2
+        x += _stroke_x_overflow(markeredgewidth, incidence_angle,
+                                vinfo.corner_angle, joinstyle, capstyle)
+        if x > extents[xmax]:
+            extents[xmax] = x
+    else:
+        if vinfo.incidence_angle < 0:  # [-pi, -pi/2]
+            incidence_angle = 3*np.pi/2 + vinfo.incidence_angle
+        else:
+            incidence_angle = vinfo.incidence_angle - np.pi/2
+        x -= _stroke_x_overflow(markeredgewidth, incidence_angle,
+                                vinfo.corner_angle, joinstyle, capstyle)
+        if x < extents[xmin]:
+            extents[xmin] = x
+    if np.sin(vinfo.incidence_angle) > 0:
+        incidence_angle = vinfo.incidence_angle
+        y += _stroke_x_overflow(markeredgewidth, incidence_angle,
+                                vinfo.corner_angle, joinstyle, capstyle)
+        if y > extents[ymax]:
+            extents[ymax] = y
+    else:
+        incidence_angle = vinfo.incidence_angle + np.pi
+        y -= _stroke_x_overflow(markeredgewidth, incidence_angle,
+                                vinfo.corner_angle, joinstyle, capstyle)
+        if y < extents[ymin]:
+            extents[ymin] = y
+    # also catch extra extent due to caps growing sideways
+    if vinfo.corner_angle is None:
+        for perp_dir in [np.pi/2, 3*np.pi/2]:
+            x, y = vinfo.apex
+            cap_perp = vinfo.incidence_angle + perp_dir
+            x += (markeredgewidth/2) * np.cos(cap_perp)
+            if x < extents[xmin]:
+                extents[xmin] = x
+            if x > extents[xmax]:
+                extents[xmax] = x
+            y += (markeredgewidth/2) * np.sin(cap_perp)
+            if y < extents[ymin]:
+                extents[ymin] = y
+            if y > extents[ymax]:
+                extents[ymax] = y
```

Comment:
I'm not sure if it looks better this way (ditto the others), or worse and slower...
```python
            cap_perp = vinfo.incidence_angle + perp_dir
            x += (markeredgewidth/2) * np.cos(cap_perp)
            extents[xmin] = min(extents[xmin], x)
            extents[xmax] = max(extents[xmax], x)
            y += (markeredgewidth/2) * np.sin(cap_perp)
            extents[ymin] = min(extents[ymin], y)
            extents[ymax] = max(extents[ymax], y)
```